src/mono/sample/iOS/Program.csproj

```diff
@@ -38,7 +38,7 @@
       <AppDir>$(MSBuildThisFileDirectory)$(PublishDir)\app</AppDir>
       <IosSimulator Condition="'$(TargetsiOSSimulator)' == 'true'">iPhone 11</IosSimulator>
       <Optimized Condition="'$(Configuration)' == 'Release'">True</Optimized>
-      <RunAOTCompilation Condition="('$(TargetsMacCatalyst)' == 'false' and '$(IosSimulator)' == '') or '$(ForceAOT)' == 'true'">true</RunAOTCompilation>
+      <RunAOTCompilation>true</RunAOTCompilation>
```

Comment:
HelloiOS sample fails on iossimulator with error `Failed to load AOT module 'System.Private.CoreLib' while running in aot-only mode: not compiled with --aot=interp`. Idea is to do AOT on a device and AOT+Interpreter on the simulators and maccatalyst. 

I wasn't sure if there are other cases we want to cover.

Reply:
> HelloiOS sample fails on iossimulator with error Failed to load AOT module 'System.Private.CoreLib' while running in aot-only mode: not compiled with --aot=interp

I am not 100% sure but I believe iossimulator-arm64 requires `-p:ForceAOT=true` 
@akoeplinger am I remember this correctly?

Reply:
I see, thanks for the explanation, these setups are tricky :) 
> The iossimulator tests are executed with /p:RunAOTCompilation=true and /p:MonoForceInterpreter=true

If that is the case, then based what was previous on line 69: 
``` xml
<AOTMode Condition="'$(TargetOS)' != 'maccatalyst'">Full</AOTMode>
``` 
it means that iossimulator tests were always using `AOTMode=Full` which further means `/p:MonoForceInterpreter=true` was ignored for them (and is redundant in that case?)

FWIW, Not sure if that is already available, but it would be great if we would have some kind of configuration matrix which would show the different setups we are using for testing, but also what are the "valid" setups (eg: targeting iOS device cannot have AOTMode=JustInterp, etc). This would help clarify how we test on devices/simulators, would avoid future confusions and in general improve maintenance. I can gladly assist if we decide something like that would be useful.

Reply:
> it means that iossimulator tests were always using `AOTMode=Full` which further means `/p:MonoForceInterpreter=true` was ignored for them (and is redundant in that case?)

I think `MonoForceInterpreter` is being used in AppleAppBuilder, but the test was failing because it was compiled with AOTMode `Full` instead of `JustInterp`. The functional tests have been disabled, and locally we weren't using the `MonoForceInterpreter` parameter.

> FWIW, Not sure if that is already available, but it would be great if we would have some kind of configuration matrix which would show the different setups we are using for testing, but also what are the "valid" setups (eg: targeting iOS device cannot have AOTMode=JustInterp, etc). This would help clarify how we test on devices/simulators, would avoid future confusions and in general improve maintenance. I can gladly assist if we decide something like that would be useful.

Good idea. Based on the CI, tests on a device are AOT compiled in `Full`AOT mode, while tests on a simulator and maccatalyst are AOT compiled in `JustInterp` AOT mode. As you said, we should update the documentation to include such matrix and make the example to AOT compile in `Full`AOT mode on a device/  `JustInterp` AOT mode on a simulator and maccatalyst. There can be an option to override `JustInterp` with `Full`AOT mode for testing locally.

Reply:
Just a thought.. 

Could we cleanup HelloiOS to use: https://github.com/dotnet/runtime/blob/7b2dc26555db2ef75c4ef822613a63b8c1d25e85/src/mono/msbuild/apple/build/AppleBuild.LocalBuild.props (or AppleBuild.props/targets) instead of all these configuration parameters being hard-coded (like `AOTMode`) in the sample project file itself?
This way the tests and sample projects would share the common specification?

To do this, we would probably have to make sure all required Items/Properties are available in AppleBuild.props and have default values based on the mentioned configuration matrix. Additionally, we would make sure that the configuration properties we want to expose could be overwritten eg: 
```xml
<AOTMode Condition="'$(AOTMode)' == ''">SomeDefaultValue</AOTMode>
```

What do you think?

Reply:
At first, I thought it might be beneficial to have MonoAOTCompiler and AppleAppBuilder in the sample project for easier debugging. However, I agree that it is better for maintenance to have a single project used across the runtime (tests and samples). 

Additionally, there are multiple occurrences of tasks in the runtime (library CI project files) that we intend to consolidate, as outlined in https://github.com/dotnet/runtime/issues/84254.